framework/py/flwr/cli/new/new.py

```diff
@@ -91,6 +103,141 @@ def render_and_create(file_path: Path, template: str, context: dict[str, str]) -
     create_file(file_path, content)
 
 
+# Security: prevent zip-slip
+def _safe_extract_zip(zf: zipfile.ZipFile, dest_dir: Path) -> None:
+    dest_dir = dest_dir.resolve()
+
+    def _is_within_directory(base: Path, target: Path) -> bool:
+        try:
+            target.relative_to(base)
+            return True
+        except ValueError:
+            return False
+
+    for member in zf.infolist():
+        # Skip directory placeholders; ZipInfo can represent them as names ending with '/'
+        if member.is_dir():
+            target_path = (dest_dir / member.filename).resolve()
+            if not _is_within_directory(dest_dir, target_path):
+                raise ValueError(f"Unsafe path in zip: {member.filename}")
+            target_path.mkdir(parents=True, exist_ok=True)
+            continue
+
+        # Files
+        target_path = (dest_dir / member.filename).resolve()
+        if not _is_within_directory(dest_dir, target_path):
+            raise ValueError(f"Unsafe path in zip: {member.filename}")
+
+        # Ensure parent exists
+        target_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Extract
+        with zf.open(member, "r") as src, open(target_path, "wb") as dst:
+            dst.write(src.read())
+
+
+def _download_zip_to_memory(presigned_url: str) -> io.BytesIO:
+    try:
+        r = requests.get(presigned_url, timeout=60)
+        r.raise_for_status()
+    except requests.RequestException as e:
+        raise typer.BadParameter(f"ZIP download failed: {e}") from e
+
+    buf = io.BytesIO(r.content)
+    # Validate it's a zip
+    if not zipfile.is_zipfile(buf):
+        raise typer.BadParameter("Downloaded file is not a valid ZIP")
+    buf.seek(0)
+    return buf
+
+
+def _request_download_link(identifier: str) -> str:
+    url = f"{PLATFORM_API_URL}/hub"
+    headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+    }
+    body = {
+        "identifier": identifier,             # send raw string of identifier
+        "flwr_version": _get_flwr_version(),  # send flwr version
+    }
+
+    try:
+        resp = requests.post(url, headers=headers, data=json.dumps(body), timeout=20)
+    except requests.RequestException as e:
+        raise typer.BadParameter(f"Failed to reach platform API: {e}") from e
+
+    if resp.status_code == 404:
+        raise typer.BadParameter("Remote app not found")
+    if not resp.ok:
+        raise typer.BadParameter(f"Platform API error: {resp.status_code} {resp.text}")
+
+    data = resp.json()
+    if "url" not in data:
+        raise typer.BadParameter("Bad response from platform API (missing 'url')")
+    return data["url"]
```

Comment:
are these `raise` handled by typer automatically? or do we need to handle them inside `download_remote_app_via_api`